### ISSUE TITLE
Object (ObjectConstructor) is wrongfully assignable to any function

### ISSUE THREAD

**Issue by szacskesz** [microsoft/TypeScript#57379]
### 🔎 Search Terms

ObjectConstructor, Object, assignable, function

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about 5.3.3

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtVJxwAowAjALniIEp4BeAPngGcMYtUBzaygNxyzAA3ACgRiQjSEB6afECg5PACiADwAO4DCGDw4zZBAyUQMGDhjjJNekwBEt6jLmLVGzNt0h9hyqhzwTMwsJYmtGeFQDCEdZBWV1TQ89AyMA03NLYgBvAF9qWJcE9x1knzSgzKIAZTYObic410SSrxTff0CMkKIAOWQAW1ITGOd4ty0W71TOi0qAeVIAK00R+EAZcjHmzynjdIsgA

### 💻 Code

```ts
declare function foo(cb: () => string): void;

foo();// ✅ Expected result: error
foo(() => "");// ✅ Expected result: no error
foo(() => null);// ✅ Expected result: error
foo({})// ✅ Expected result: error
foo(String);// ✅ Expected result: no error
foo(Number);// ✅ Expected result: error

foo(Object);// ❌ Expected result: error
```

### 🙁 Actual behavior

`Object` (`ObjectContructor`) is assignable to any function, because it has a callable signature with return type of any.

This lets the developers to use `Object` where only a properly typed function would be allowed, which can lead to unexpected runtime errors.

I know this is a pretty edge case, but in my opinion it's worth looking into.

### 🙂 Expected behavior

`Object` is not assignable to functions with different return type

**Comment by RyanCavanaugh** [comment]
It's declared as having a `(): any;` call signature since the definition predates the existence of `unknown`/`object` and types deriving explicitly from `Object` are obviously free to be ~anything. I don't see much upside in changing that -- this is basically the only observable effect of that.

**Comment by rotu** [comment]
I think `object` would be more appropriate than `unknown`. The return type of `Object` is known to not be a primitive type and can be used where an Object is required, e.g. as the key of a `WeakMap`. See also e.g. #57000

Also, it's confusing that `Object()` and `new Object()` have different return types. Not sure if intentional.
